### PR TITLE
Support COMMAND [...] 

### PR DESCRIPTION
**Pull request by shacharPash**
Add support to commands:
COMMAND COUNT - https://redis.io/commands/command-count/
COMMAND GETKEYS - https://redis.io/commands/command-getkeys/
COMMAND LIST - https://redis.io/commands/command-list/

### CODE REVIEW

**Comment by NickCraver** [comment]
I just see some trailing spaces on tests to tidy - wanted to comment early this is looking great!

**Comment by shacharPash** [comment]
@NickCraver 
Ready

**Comment by chayim** [comment]
Part of #2055

**Review by NickCraver — CHANGES_REQUESTED** [review]
Trying to catch up a bit - thanks for this! The first set looks good, but I think the second set is split up a bit too much - normally in our API patterns you'd expect 1 command with some arguments for `CommandGetkeys`, so let's collapse that down to the main `CommandGetkeys()` method with string params for the 3 possible filters :)

File: src/StackExchange.Redis/Interfaces/IServer.cs [hunk 1]
```diff
@@ -179,6 +179,70 @@ public partial interface IServer : IRedis
...
+        /// <inheritdoc cref="CommandCount(CommandFlags)"/>
+        Task<long> CommandCountAsync(CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Returns List of keys from a full Redis command.
+        /// </summary>
+        /// <param name="command">The Command.</param>
+        /// <param name="flags">The command flags to use.</param>
+        /// <remarks><seealso href="https://redis.io/commands/command-getkeys"/></remarks>
+        RedisValue[] CommandGetkeys(RedisValue[] command, CommandFlags flags = CommandFlags.None);
```
[the remark this answers is not in the document]
Comment (shacharPash) [reply src/StackExchange.Redis/Interfaces/IServer.cs:198]: string[] will be better?

File: src/StackExchange.Redis/Interfaces/IServer.cs [hunk 2]
```diff
@@ -179,6 +179,40 @@ public partial interface IServer : IRedis
...
+        /// </summary>
+        /// <param name="moduleName">The command list will filtered by this Module Name.</param>
+        /// <param name="category">The command list will filtered by this category.</param>
+        /// <param name="pattern">The command list will filtered by this pattern.</param>
+        /// <param name="flags">The command flags to use.</param>
+        /// <remarks><seealso href="https://redis.io/commands/command-list"/></remarks>
+        RedisValue[] CommandList(RedisValue? moduleName = null, RedisValue? category = null, RedisValue? pattern = null, CommandFlags flags = CommandFlags.None);
+
+        /// <inheritdoc cref="CommandList(RedisValue?, RedisValue?, RedisValue?, CommandFlags)"/>
+        Task<RedisValue[]> CommandListAsync(RedisValue? moduleName = null, RedisValue? category = null, RedisValue? pattern = null, CommandFlags flags = CommandFlags.None);
```
Comment (NickCraver) [inline src/StackExchange.Redis/Interfaces/IServer.cs:214]: Let's make this return `string[]` please :)

File: src/StackExchange.Redis/Interfaces/IServer.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (NickCraver) [reply src/StackExchange.Redis/Interfaces/IServer.cs]: Pushed some tweaks - all good here now!

File: src/StackExchange.Redis/Interfaces/IServer.cs (hunk 2 above)
Comment (NickCraver) [reply src/StackExchange.Redis/Interfaces/IServer.cs]: Done!

**Pull request merged**